src/wombats/lib/wombat_lib.clj

```diff
@@ -0,0 +1,232 @@
+(def turn-directions [:right :left :about-face])
+(def smoke-directions [:forward :backward :left :right :drop])
+(def game-parameters
+{;; HP Modifiers
+ :collision-hp-damage 10
+ :food-hp-bonus 5
+ :poison-hp-damage 10
+ ;; Score Modifiers
+ :food-score-bonus 10
+ :wombat-hit-bonus 10
+ :zakano-hit-bonus 8
+ :steel-barrier-hit-bonus 2
+ :wood-barrier-hit-bonus 2
+ :wombat-destroyed-bonus 25
+ :zakano-destroyed-bonus 15
+ :wood-barrier-destroyed-bonus 3
+ :steel-barrier-destroyed-bonus 25
+ ;; In game parameters
+ :shot-distance 5})
+
+(defn get-arena-size
+  "Fetchs the size of one side of the arena from the state"
+  [state]
+  (first (:global-dimensions state)))
+
+(defn in?
+  "Return true if coll contains elem"
+  [elem coll]
+  (some #(= elem %) coll))
+
+(defn add-locs
+  "Add local :x and :y coordinates to state matrix"
+  [arena]
+  (reduce
+    #(conj %1
+      (reduce
+        (fn [acc node] (conj acc (assoc node :x (count acc) :y (count %1))))
+        [] %2))
+    [] arena))
+
+(defn filter-arena
+  "Filter the arena to return only nodes that contain one of the given type"
+  ([arena] (flatten arena))
+  ([arena & filters]
+  (let [node-list (flatten arena)]
+    (filter #(in? (get-in % [:contents :type]) filters) node-list))))
+
+(defn build-initial-global-state
+  "Constructs an initial global state populated by fog"
+  [global-size]
+  (add-locs (into [] (map (fn [_] 
+     (into [] (map (fn [_] {:type "fog"}) (range global-size)))) (range global-size)))))
+
+(defn add-to-state
+  "Update the global saved state with the given element and position"
+  [matrix elem]
```

Comment:
Probably would be cleaner to destructure in the args list so you don't need the `let` block.

```
(defn add-to-state
  [matrix {:keys [x y] :as elem}]
  (assoc matrix y (assoc (nth matrix y) x elem))